app/api/resources/grpc_client.py

```diff
@@ -60,4 +60,15 @@ def get_popular_species(self):
         popular_species_request = ensembl_metadata_pb2.OrganismsGroupRequest()
         popular_species = self.stub.GetOrganismsGroupCount(popular_species_request)
 
-        return popular_species
\ No newline at end of file
+        return popular_species
+
+    def get_karyotype(self, genome_uuid: str):
+        genome_assembly_request = ensembl_metadata_pb2.GenomeAssemblySequenceRequest(
+            genome_uuid=genome_uuid,
+            chromosomal_only=True
+            )
+        karyotypes = self.stub.GetGenomeAssemblySequence(genome_assembly_request)
```

Comment:
Just an observation (it is entirely up to you how you name things, but in case you find this useful) — a karyotype is a set of chromosomes. So what you are retrieving from the gRPC service is not karyotypes (plural), but a karyotype (singular), which consists of chromosomes, or top-level regions (this means the same in this context). What you are then iterating over in the for loop below is not karyotypes, but chromosomes/regions. One dictionary that you create represents not a karyotype, but a chromosome/region. It is the array of those dictionaries that you eventually send to the client that is the karyotype.

Reply:
Makes sense. It has now been updated.